table/format.h

```diff
@@ -128,66 +129,46 @@ inline bool IsSupportedFormatVersion(uint32_t version) {
 // elsewhere under the metaindex block. For example, checksum_type is
 // required for verifying metaindex block checksum (when applicable), but
 // index block handle can easily go in metaindex block (possible future).
+// See also FooterBuilder below.
 class Footer {
  public:
+  // Create empty. Populate using DecodeFrom.
   Footer() {}
 
-  // Uses builder pattern rather than distinctive ctors
+  // Deserialize a footer (populate fields) from `input` and check for various
+  // corruptions. On success (and some error cases) `input` is advanced past
+  // the footer. `input_offset` is the offset within the target file of
+  // `input` buffer (future use).
+  Status DecodeFrom(Slice input, uint64_t input_offset);
 
```

Comment:
This is non-intuitive C++ coding style (at least to me).  This takes a Slice (not Slice* or Slice&) and appears to alter the input value (by removing the prefix, etc).  The comment implies it alters input, which means the value is not really "input" but at best "input/output". 

I would prefer this be passed in as Slice&, to indicate the value could be changed and show the expectations of it being modified.

Reply:
I forgot to update the comment.

The purpose of a footer is to come at the end of a file. Thus naturally (and actually) the caller doesn't care about advancing the input slice for parsing footers. The old `Mark all input consumed` code did not do anything useful and likely impacted performance, along with the indirection. You have to have good reasons for keeping/adding currently unnecessary code and complexity to internal APIs. (In/out param where in suffices is more complexity.)

I was forced into caring about these small bits of performance by the benchmark results. Someone can always refactor this function again if they need it to do something else.